src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/api/user.ts

```diff
@@ -2,8 +2,28 @@ import { getClient } from "@/api/client";
 import type { Components } from "@/types/openapi";
 
 const getUserInfo = async (): Promise<Components.Schemas.UserProfile | null> => {
-  const client = await getClient();
-  return (await client.GetUserInfo()).data?.userInfo ?? null;
+  try {
+    const client = await getClient();
+    const response = await client.GetUserInfo();
+
+    // Check if the response has data and userInfo property
+    return response?.data?.userInfo ?? null;
+  } catch (error) {
```

Comment:
should it be more specific catch on 404?

Reply:
Yes can do. It doesn't change much as far as handling the error at this point. We still need to identify how we want to handle errors FE moving forward (alert/toast user, log?, etc.). 

Adjusted to add catch `404`

Reply:
not exactly - 404 here has a true meaning: that user was not found, hence it is a new user